hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -686,7 +686,7 @@ public static Stream<HoodieRecord> createPartitionStatsRecords(String partitionP
 
       HoodieMetadataPayload payload = new HoodieMetadataPayload(key.getRecordKey(),
           HoodieMetadataColumnStats.newBuilder()
-              .setFileName(null)
+              .setFileName(new StoragePath(columnRangeMetadata.getFilePath()).getName())
```

Comment:
Is this index used to fileter candidate files?
in this code, the partition stats is a aggrate metric, not belong to a certain file
```scala
// Step 1: Collect Column Metadata for Each File
      List<List<HoodieColumnRangeMetadata<Comparable>>> fileColumnMetadata = partitionInfo.getFileNameToSizeMap().keySet().stream()
          .map(fileName -> getFileStatsRangeMetadata(partitionPath, partitionPath + "/" + fileName, dataTableMetaClient, columnsToIndex, false))
          .collect(toList());
      // Step 2: Flatten and Group by Column Name
      Map<String, List<HoodieColumnRangeMetadata<Comparable>>> columnMetadataMap = fileColumnMetadata.stream()
          .flatMap(List::stream)
          .collect(Collectors.groupingBy(HoodieColumnRangeMetadata::getColumnName, toList())); // Group by column name
      // Step 3: Aggregate Column Ranges
      Stream<HoodieColumnRangeMetadata<Comparable>> partitionStatsRangeMetadata = columnMetadataMap.entrySet().stream()
          .map(entry -> FileFormatUtils.getColumnRangeInPartition(entry.getValue()));
      return HoodieMetadataPayload.createPartitionStatsRecords(partitionPath, partitionStatsRangeMetadata.collect(toList()), false).iterator();
```
and in this code, filepath still be null
```scala
private static <T extends Comparable<T>> HoodieColumnRangeMetadata<T> mergeRanges(HoodieColumnRangeMetadata<T> one,
                                                                                    HoodieColumnRangeMetadata<T> another) {
    ValidationUtils.checkArgument(one.getColumnName().equals(another.getColumnName()),
        "Column names should be the same for merging column ranges");
    final T minValue = getMinValueForColumnRanges(one, another);
    final T maxValue = getMaxValueForColumnRanges(one, another);

    return HoodieColumnRangeMetadata.create(
        null, one.getColumnName(), minValue, maxValue,
        one.getNullCount() + another.getNullCount(),
        one.getValueCount() + another.getValueCount(),
        one.getTotalSize() + another.getTotalSize(),
        one.getTotalUncompressedSize() + another.getTotalUncompressedSize());
  }
```

Reply:
@KnightChess Thanks for the review. This behaves just like colstats at partition level. You can try out the test that I have added from spark shell and on Spark UI, it should show less number of files scanned. I will fix the issue you're pointing out. Ideally, we need a flag in `ColumnStatsIndexSupport` to indicate that candidate file filtering is happening for partition stats.